pkg/storage/oss/oss.go

```diff
@@ -0,0 +1,673 @@
+package oss
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"reflect"
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/auth"
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/auth/credentials"
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/auth/credentials/provider"
+	"github.com/aliyun/aliyun-oss-go-sdk/oss"
+
+	configv1 "github.com/openshift/api/config/v1"
+	imageregistryv1 "github.com/openshift/api/imageregistry/v1"
+	operatorapi "github.com/openshift/api/operator/v1"
+
+	regopclient "github.com/openshift/cluster-image-registry-operator/pkg/client"
+	"github.com/openshift/cluster-image-registry-operator/pkg/defaults"
+	"github.com/openshift/cluster-image-registry-operator/pkg/envvar"
+	"github.com/openshift/cluster-image-registry-operator/pkg/storage/util"
+)
+
+const (
+	imageRegistrySecretMountpoint = "/var/run/secrets/cloud"
+	imageRegistrySecretDataKey    = "credentials"
+	imageRegistryAccessKeyID      = "access_key_id"
+	imageRegistryAccessKeySecret  = "access_key_secret"
+
+	envRegistryStorage                   = "REGISTRY_STORAGE"
+	envRegistryStorageOssBucket          = "REGISTRY_STORAGE_OSS_BUCKET"
+	envRegistryStorageOssRegion          = "REGISTRY_STORAGE_OSS_REGION"
+	envRegistryStorageOssEndpoint        = "REGISTRY_STORAGE_OSS_ENDPOINT"
+	envRegistryStorageOssEncrypt         = "REGISTRY_STORAGE_OSS_ENCRYPT"
+	envRegistryStorageOssInternal        = "REGISTRY_STORAGE_OSS_INTERNAL"
+	envRegistryStorageOssAccessKeyId     = "REGISTRY_STORAGE_OSS_ACCESSKEYID"
+	envRegistryStorageOssAccessKeySecret = "REGISTRY_STORAGE_OSS_ACCESSKEYSECRET"
```

Comment:
These constants are used only once, please inline them into the code.

Reply:
ok